### ISSUE TITLE
JDK8 Unhandled trap vmState=0x0005ff0b

### ISSUE THREAD

**Issue by connglli** [eclipse-openj9/openj9#15599]
Java -version output
--------------------
```
openjdk version "1.8.0_342-internal"
OpenJDK Runtime Environment (build 1.8.0_342-internal-_2022_06_10_15_18-b00)
Eclipse OpenJ9 VM (build master-3d06b2f9c, JRE 1.8.0 Linux amd64-64-Bit Compressed References 20220610_000000 (JIT enabled, AOT enabled)
OpenJ9   - 3d06b2f9c
OMR      - cf8ddbd1a
JCL      - 2bb179375a based on jdk8u342-b05)
```

Summary of problem
------------------

A new one, looks quite similar to #15305, but different crash: `Type=Unhandled trap vmState=0x0005ff0b`.

``` java
class Test {
  int b;
  long c;
  int j;
  float d;
  void e() {
    for (int ax$3 = -4605; ax$3 < 2;)
      try {
        double[] ax$0 = new double[1];
        ax$0[0] = 0;
      } catch (Throwable $) {
      }
  }
  int g(long h, float f) {
    int k[] = new int[b];
    e();
    long l = m(k);
    return (int)l;
  }
  void n(int i) { g(c, d); }
  void r(String[] o) { n(j); }
  public static void main(String[] p) {
    Test s = new Test();
    for (;;)
      s.r(p);
  }
  long m(int[] a) {
    long q = 0;
    return q;
  }
}
```

Diagnostic files
----------------

By issuing 

```
$ java -Xmx1G Test
```

the following crash log is given:

```
Assertion failed at /root/hostdir/openj9-openjdk-jdk8/omr/compiler/codegen/Relocation.cpp:68: _label->getCodeLocation() != NULL
VMState: 0x0005ff0b
  cannot relocate reference to undefined label: <unknown labelsym> (0x7fa0807211e0)

compiling Test.e()V at level: very-hot (profiling)
#0: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x7ab6d5) [0x7fa0834a96d5]
#1: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x7b69c0) [0x7fa0834b49c0]
#2: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x46de64) [0x7fa08316be64]
#3: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x46f61d) [0x7fa08316d61d]
#4: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3edf44) [0x7fa0830ebf44]
#5: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3ee151) [0x7fa0830ec151]
#6: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3f94dd) [0x7fa0830f74dd]
#7: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0xdfe46) [0x7fa082ddde46]
#8: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3ff98d) [0x7fa0830fd98d]
#9: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x4019d9) [0x7fa0830ff9d9]
#10: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3fe623) [0x7fa0830fc623]
#11: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x42bb29) [0x7fa083129b29]
#12: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x121e6a) [0x7fa082e1fe6a]
#13: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x122c73) [0x7fa082e20c73]
#14: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9prt29.so(+0x29c53) [0x7fa088db9c53]
#15: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x120a69) [0x7fa082e1ea69]
#16: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x1210d7) [0x7fa082e1f0d7]
#17: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x11fdab) [0x7fa082e1ddab]
#18: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x12009a) [0x7fa082e1e09a]
#19: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x120162) [0x7fa082e1e162]
#20: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9prt29.so(+0x29c53) [0x7fa088db9c53]
#21: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x1205a2) [0x7fa082e1e5a2]
#22: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9thr29.so(+0xf2b2) [0x7fa088f7c2b2]
#23: /lib/x86_64-linux-gnu/libpthread.so.0(+0x8609) [0x7fa089222609]
#24: /lib/x86_64-linux-gnu/libc.so.6(clone+0x43) [0x7fa089362133]

#0: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x7ab6d5) [0x7fa0834a96d5]
#1: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x7b69c0) [0x7fa0834b49c0]
#2: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x1125d9) [0x7fa082e105d9]
#3: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9prt29.so(+0x2911a) [0x7fa088db911a]
#4: /lib/x86_64-linux-gnu/libpthread.so.0(+0x14420) [0x7fa08922e420]
#5: /lib/x86_64-linux-gnu/libc.so.6(gsignal+0xcb) [0x7fa08928600b]
#6: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x46f597) [0x7fa08316d597]
#7: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x46f622) [0x7fa08316d622]
#8: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3edf44) [0x7fa0830ebf44]
#9: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3ee151) [0x7fa0830ec151]
#10: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3f94dd) [0x7fa0830f74dd]
#11: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0xdfe46) [0x7fa082ddde46]
#12: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3ff98d) [0x7fa0830fd98d]
#13: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x4019d9) [0x7fa0830ff9d9]
#14: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3fe623) [0x7fa0830fc623]
#15: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x42bb29) [0x7fa083129b29]
#16: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x121e6a) [0x7fa082e1fe6a]
#17: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x122c73) [0x7fa082e20c73]
#18: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9prt29.so(+0x29c53) [0x7fa088db9c53]
#19: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x120a69) [0x7fa082e1ea69]
#20: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x1210d7) [0x7fa082e1f0d7]
#21: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x11fdab) [0x7fa082e1ddab]
#22: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x12009a) [0x7fa082e1e09a]
#23: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x120162) [0x7fa082e1e162]
#24: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9prt29.so(+0x29c53) [0x7fa088db9c53]
#25: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x1205a2) [0x7fa082e1e5a2]
#26: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9thr29.so(+0xf2b2) [0x7fa088f7c2b2]
#27: /lib/x86_64-linux-gnu/libpthread.so.0(+0x8609) [0x7fa089222609]
#28: /lib/x86_64-linux-gnu/libc.so.6(clone+0x43) [0x7fa089362133]
Unhandled exception
Type=Unhandled trap vmState=0x0005ff0b
J9Generic_Signal_Number=00000108 Signal_Number=00000005 Error_Value=00000000 Signal_Code=fffffffa
Handler1=00007FA088FD0020 Handler2=00007FA088DB8EF0
RDI=0000000000000002 RSI=00007FA0817B2760 RAX=0000000000000000 RBX=00007FA0807211E0
RCX=00007FA08928600B RDX=0000000000000000 R8=0000000000000000 R9=00007FA0817B2760
R10=0000000000000008 R11=0000000000000246 R12=00007FA0805BEB08 R13=00007FA0805BA5A0
R14=00007FA083621D3A R15=00007FA0817B2AC0
RIP=00007FA08928600B GS=0000 FS=0000 RSP=00007FA0817B2760
EFlags=0000000000000246 CS=0033 RBP=00007FA0805BE7F0 ERR=0000000000000000
TRAPNO=0000000000000000 OLDMASK=0000000000000000 CR2=0000000000000000
xmm0 ffffffffffffffff (f: 4294967296.000000, d: -nan)
xmm1 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm2 000000000000ff00 (f: 65280.000000, d: 3.225261e-319)
xmm3 00007fa0817b2230 (f: 2172330496.000000, d: 6.933092e-310)
xmm4 4000000000000000 (f: 0.000000, d: 2.000000e+00)
xmm5 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm6 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm7 00007fa0817b9ef0 (f: 2172362496.000000, d: 6.933092e-310)
xmm8 6f6e6e612f3c000a (f: 792461312.000000, d: 5.767205e+228)
xmm9 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
Module=/lib/x86_64-linux-gnu/libc.so.6
Module_base_address=00007FA089243000 Symbol=gsignal
Symbol_address=00007FA089285F40

Method_being_compiled=Test.e()V
Target=2_90_20220610_000000 (Linux 5.4.0-122-generic)
CPU=amd64 (8 logical CPUs) (0x3e45bc000 RAM)
----------- Stack Backtrace -----------
gsignal+0xcb (0x00007FA08928600B [libc.so.6+0x4300b])
_ZN2TR4trapEv+0x47 (0x00007FA08316D597 [libj9jit29.so+0x46f597])
 (0x00007FA08316D622 [libj9jit29.so+0x46f622])
 (0x00007FA0830EBF44 [libj9jit29.so+0x3edf44])
_ZN2TR28LabelRelative32BitRelocation5applyEPNS_13CodeGeneratorE+0x11 (0x00007FA0830EC151 [libj9jit29.so+0x3ee151])
_ZN3OMR13CodeGenerator18processRelocationsEv+0x2d (0x00007FA0830F74DD [libj9jit29.so+0x3f94dd])
_ZN2J913CodeGenerator18processRelocationsEv+0x16 (0x00007FA082DDDE46 [libj9jit29.so+0xdfe46])
_ZN3OMR12CodeGenPhase30performProcessRelocationsPhaseEPN2TR13CodeGeneratorEPNS1_12CodeGenPhaseE+0x7d (0x00007FA0830FD98D [libj9jit29.so+0x3ff98d])
_ZN3OMR12CodeGenPhase10performAllEv+0xc9 (0x00007FA0830FF9D9 [libj9jit29.so+0x4019d9])
_ZN3OMR13CodeGenerator12generateCodeEv+0x63 (0x00007FA0830FC623 [libj9jit29.so+0x3fe623])
_ZN3OMR11Compilation7compileEv+0xa29 (0x00007FA083129B29 [libj9jit29.so+0x42bb29])
_ZN2TR28CompilationInfoPerThreadBase7compileEP10J9VMThreadPNS_11CompilationEP17TR_ResolvedMethodR11TR_J9VMBaseP19TR_OptimizationPlanRKNS_16SegmentAllocatorE+0x4fa (0x00007FA082E1FE6A [libj9jit29.so+0x121e6a])
_ZN2TR28CompilationInfoPerThreadBase14wrappedCompileEP13J9PortLibraryPv+0x323 (0x00007FA082E20C73 [libj9jit29.so+0x122c73])
omrsig_protect+0x1e3 (0x00007FA088DB9C53 [libj9prt29.so+0x29c53])
_ZN2TR28CompilationInfoPerThreadBase7compileEP10J9VMThreadP21TR_MethodToBeCompiledRN2J917J9SegmentProviderE+0x309 (0x00007FA082E1EA69 [libj9jit29.so+0x120a69])
_ZN2TR24CompilationInfoPerThread12processEntryER21TR_MethodToBeCompiledRN2J917J9SegmentProviderE+0x207 (0x00007FA082E1F0D7 [libj9jit29.so+0x1210d7])
_ZN2TR24CompilationInfoPerThread14processEntriesEv+0x38b (0x00007FA082E1DDAB [libj9jit29.so+0x11fdab])
_ZN2TR24CompilationInfoPerThread3runEv+0x2a (0x00007FA082E1E09A [libj9jit29.so+0x12009a])
_Z30protectedCompilationThreadProcP13J9PortLibraryPN2TR24CompilationInfoPerThreadE+0x82 (0x00007FA082E1E162 [libj9jit29.so+0x120162])
omrsig_protect+0x1e3 (0x00007FA088DB9C53 [libj9prt29.so+0x29c53])
_Z21compilationThreadProcPv+0x1d2 (0x00007FA082E1E5A2 [libj9jit29.so+0x1205a2])
thread_wrapper+0x162 (0x00007FA088F7C2B2 [libj9thr29.so+0xf2b2])
start_thread+0xd9 (0x00007FA089222609 [libpthread.so.0+0x8609])
clone+0x43 (0x00007FA089362133 [libc.so.6+0x11f133])
---------------------------------------
JVMDUMP039I Processing dump event "gpf", detail "" at 2022/07/21 18:30:47 - please wait.
JVMDUMP032I JVM requested System dump using '/zdata/congli/ax-exp/ax-eval/2-ax-only/62.openj9/mutant/red/ttt/red/core.20220721.183047.2748928.0001.dmp' in response to an event
JVMDUMP010I System dump written to /zdata/congli/ax-exp/ax-eval/2-ax-only/62.openj9/mutant/red/ttt/red/core.20220721.183047.2748928.0001.dmp
JVMDUMP032I JVM requested Java dump using '/zdata/congli/ax-exp/ax-eval/2-ax-only/62.openj9/mutant/red/ttt/red/javacore.20220721.183047.2748928.0002.txt' in response to an event
JVMDUMP010I Java dump written to /zdata/congli/ax-exp/ax-eval/2-ax-only/62.openj9/mutant/red/ttt/red/javacore.20220721.183047.2748928.0002.txt
JVMDUMP032I JVM requested Snap dump using '/zdata/congli/ax-exp/ax-eval/2-ax-only/62.openj9/mutant/red/ttt/red/Snap.20220721.183047.2748928.0003.trc' in response to an event
JVMDUMP010I Snap dump written to /zdata/congli/ax-exp/ax-eval/2-ax-only/62.openj9/mutant/red/ttt/red/Snap.20220721.183047.2748928.0003.trc
JVMDUMP032I JVM requested JIT dump using '/zdata/congli/ax-exp/ax-eval/2-ax-only/62.openj9/mutant/red/ttt/red/jitdump.20220721.183047.2748928.0004.dmp' in response to an event
JVMDUMP051I JIT dump occurred in 'JIT Compilation Thread-000' thread 0x000000000001E600
JVMDUMP049I JIT dump notified all waiting threads of the current method to be compiled
JVMDUMP054I JIT dump is tracing the IL of the method on the crashed compilation thread
JVMDUMP048I JIT dump method being compiled is an ordinary method
JVMDUMP053I JIT dump is recompiling Test.e()V
Assertion failed at /root/hostdir/openj9-openjdk-jdk8/omr/compiler/codegen/Relocation.cpp:68: _label->getCodeLocation() != NULL
VMState: 0x0005ff0b
  cannot relocate reference to undefined label: Label L0019 (0x7fa06e3958f0)

compiling Test.e()V at level: very-hot (profiling)
#0: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x7ab6d5) [0x7fa0834a96d5]
#1: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x7b69c0) [0x7fa0834b49c0]
#2: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x46de64) [0x7fa08316be64]
#3: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x46f61d) [0x7fa08316d61d]
#4: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3edf44) [0x7fa0830ebf44]
#5: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3ee151) [0x7fa0830ec151]
#6: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3f94dd) [0x7fa0830f74dd]
#7: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0xdfe46) [0x7fa082ddde46]
#8: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3ff98d) [0x7fa0830fd98d]
#9: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x4019d9) [0x7fa0830ff9d9]
#10: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x3fe623) [0x7fa0830fc623]
#11: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x42bb29) [0x7fa083129b29]
#12: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x121e6a) [0x7fa082e1fe6a]
#13: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x122c73) [0x7fa082e20c73]
#14: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9prt29.so(+0x29c53) [0x7fa088db9c53]
#15: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x120b7a) [0x7fa082e1eb7a]
#16: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x1210d7) [0x7fa082e1f0d7]
#17: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x11fdab) [0x7fa082e1ddab]
#18: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x12009a) [0x7fa082e1e09a]
#19: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x120162) [0x7fa082e1e162]
#20: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9prt29.so(+0x29c53) [0x7fa088db9c53]
#21: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9jit29.so(+0x1205a2) [0x7fa082e1e5a2]
#22: /zdata/congli/OpenJ9/jdk8/jre/lib/amd64/default/libj9thr29.so(+0xf2b2) [0x7fa088f7c2b2]
#23: /lib/x86_64-linux-gnu/libpthread.so.0(+0x8609) [0x7fa089222609]
#24: /lib/x86_64-linux-gnu/libc.so.6(clone+0x43) [0x7fa089362133]

JVMDUMP052I JIT dump recursive crash occurred on diagnostic thread
JVMDUMP010I JIT dump written to /zdata/congli/ax-exp/ax-eval/2-ax-only/62.openj9/mutant/red/ttt/red/jitdump.20220721.183047.2748928.0004.dmp
JVMDUMP013I Processed dump event "gpf", detail "".
```

Please also check [openj9-bug-112.tar.gz](https://drive.google.com/file/d/1Ja1QHt5lJ2G_3lB29-5zJ7Ujeg53CQJ7/view?usp=sharing) for all the logs (jitdump, snap, etc.) and the testcase (Test.java, Test.class).

**Comment by 0xdaryl** [comment]
This kind of crash usually occurs when someone creates a LabelSymbol to branch to but doesn't emit a LABEL instruction to anchor it.  Please investigate @BradleyWood .

**Comment by 0xdaryl** [comment]
This will not be fixed for 0.35.  Moving to 0.36.

**Comment by 0xdaryl** [comment]
I'm going to leave this targeting 0.36 for now.  If the problem is what I suspected above then these kinds of problems are generally straightforward to get to the bottom of once identified.  @BradleyWood, please triage and report whether this can be fixed in the 0.36 timeframe.

**Comment by BradleyWood** [comment]
@0xdaryl This is the same issue as #15305 which was resolved in #6576.

**Comment by 0xdaryl** [comment]
Thanks for confirming @BradleyWood.  Closing.
